tests/acceptance/features/webUISharingPermissions/sharePermissionsUsers.feature

```diff
@@ -233,27 +233,245 @@ Feature: Sharing files and folders with internal users with different permission
+    And user "user1" has logged in using the webUI
+    When the user shares folder "simple-folder (2)" with user "User Three" as "Custom role" with permissions "share, delete" using the webUI
+    And the user re-logs in as "user3" using the webUI
+    Then the collaborators list for folder "simple-folder (2)" should be empty
+    And user "user3" should have received a share with these details:
+      | field       | value               |
+      | uid_owner   | user1               |
+      | share_with  | user3               |
+      | file_target | /simple-folder (2)  |
+      | item_type   | folder              |
+      | permissions | share, delete, read |
+
+  Scenario: User is allowed to reshare a file/folder with the lesser permissions, and check if it is listed for original owner
+    Given user "user3" has been created with default attributes
+    And user "user2" has shared folder "simple-folder" with user "user1" with "read, share, delete" permissions
+    And user "user1" has logged in using the webUI
+    When the user shares folder "simple-folder (2)" with user "User Three" as "Custom role" with permissions "delete" using the webUI
+    And the user re-logs in as "user2" using the webUI
+    Then user "User One" should be listed as "Custom role" in the collaborators list for folder "simple-folder" on the webUI
+    And custom permissions "share, delete" should be set for user "User One" for folder "simple-folder" on the webUI
+    And user "User Three" should be listed as "Custom role" in the collaborators list for folder "simple-folder" on the webUI
+    And custom permissions "delete" should be set for user "User Three" for folder "simple-folder" on the webUI
+    And user "user3" should have received a share with these details:
+      | field       | value              |
+      | uid_owner   | user1              |
+      | share_with  | user3              |
+      | file_target | /simple-folder (2) |
+      | item_type   | folder             |
+      | permissions | delete, read       |
+
+  Scenario: User is not allowed to reshare a file/folder with the higher permissions
+    Given user "user3" has been created with default attributes
+    And user "user2" has shared folder "simple-folder" with user "user1" with "read, share, delete" permissions
+    And user "user1" has logged in using the webUI
+    When the user shares folder "simple-folder (2)" with user "User Three" as "Custom role" with permissions "share, delete, change" using the webUI
+    Then the error message "Error while sharing." should be displayed on the webUI
+    When the user re-logs in as "user2" using the webUI
+    Then user "User One" should be listed as "Custom role" in the collaborators list for folder "simple-folder" on the webUI
+    And custom permissions "share, delete" should be set for user "User One" for folder "simple-folder" on the webUI
+    And user "User Three" should not be listed in the collaborators list on the webUI
```

Comment:
replace by an API step checking that no sharing happened

Reply:
Replaced with assertion that `"simple-folder (2)" should not exist`